modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/DistributionZonesUtil.java

```diff
@@ -347,4 +349,20 @@ public static DistributionZoneConfiguration getZoneById(DistributionZonesConfigu
 
         throw new DistributionZoneNotFoundException(zoneId);
     }
+
+    /**
+     * Check if a passed filter is a valid {@link JsonPath} query.
+     *
+     * @param filter Filter.
+     * @return {@code true} if the passed filter is a valid filter, {@code false} otherwise.
+     */
+    public static boolean validate(String filter) {
+        try {
+            JsonPath.compile(filter);
+        } catch (InvalidPathException ignored) {
```

Comment:
What about validation details? It's useful to propagate validation details to the user.

Reply:
added